llvm_util/llvm2alive.cpp

```diff
@@ -93,9 +94,20 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
     return i;
   }
 
-  auto get_operand(llvm::Value *v) {
+
+  Value* copy_inserter(AggregateValue *ag) {
+    auto v = make_unique<UnaryOp>(*const_cast<Type *>(&ag->getType()),
+                                  "%__copy_" + to_string(copy_idx++), *ag,
+                                  UnaryOp::Copy);
+    auto val = v.get();
+    BB->addInstr(move(v));
+    return val;
+  }
+
+  auto get_operand(llvm::Value *v, bool enable_copy_inserter = true) {
     return llvm_util::get_operand(v,
-        [this](auto I) { return convert_constexpr(I); });
+        [this](auto I) { return convert_constexpr(I); },
+        [&](auto ag) { return enable_copy_inserter ? copy_inserter(ag) : ag; });
```

Comment:
it seems what we want is to introduce copies only when the aggregate is possibly non constant. Can't we detect that instead of special casing shuffle vector?